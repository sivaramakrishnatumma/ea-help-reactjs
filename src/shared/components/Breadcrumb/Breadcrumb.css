.breadcrumb {
  background-color: rgba(22,22,22,0.6) !important;
    color: #f3f3f3;
    border-radius: 0px !important;
    margin-bottom: 0 !important;
}

.breadcrumb-item.active, .breadcrumb-item, .breadcrumb a {
  color: #f3f3f3 !important;
}

.breadcrumb-item + .breadcrumb-item::before {
  color: #f3f3f3 !important;
}