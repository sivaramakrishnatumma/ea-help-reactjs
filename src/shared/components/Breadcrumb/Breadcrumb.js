import React from "react";
import { Link } from "react-router-dom";
import "./Breadcrumb.css";

const Breadcrumb = props => {
  const items = props.items.map((item, index) => {
    return (
      <li className="breadcrumb-item" key={index}>
        <Link to={item.path}>{item.name}</Link>
      </li>
    );
  });
  return (
    <nav aria-label="breadcrumb">
      <ol className="breadcrumb">
        {items}
        <li className="breadcrumb-item active" aria-current="page">
          {props.currentPage}
        </li>
      </ol>
    </nav>
  );
};

export default Breadcrumb;
