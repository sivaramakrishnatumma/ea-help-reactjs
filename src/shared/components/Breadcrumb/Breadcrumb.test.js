import React from "react";
import { mount } from "enzyme";
import Breadcrumb from "./Breadcrumb";
import { BrowserRouter } from "react-router-dom";

describe("Breadcrumb component test case", () => {
  it("should test Breadcrumb component with default props", () => {
    const props = {
      items: [{ name: "Home", path: "/" }],
      currentPage: "Case Information"
    };
    const wrapper = mount(
      <BrowserRouter>
        <Breadcrumb {...props} />
      </BrowserRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
