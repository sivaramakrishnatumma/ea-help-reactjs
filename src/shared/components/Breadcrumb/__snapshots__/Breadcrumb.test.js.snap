// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component test case should test Breadcrumb component with default props 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Breadcrumb
      currentPage="Case Information"
      items={
        Array [
          Object {
            "name": "Home",
            "path": "/",
          },
        ]
      }
    >
      <nav
        aria-label="breadcrumb"
      >
        <ol
          className="breadcrumb"
        >
          <li
            className="breadcrumb-item"
            key="0"
          >
            <Link
              to="/"
            >
              <LinkAnchor
                href="/"
                navigate={[Function]}
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  Home
                </a>
              </LinkAnchor>
            </Link>
          </li>
          <li
            aria-current="page"
            className="breadcrumb-item active"
          >
            Case Information
          </li>
        </ol>
      </nav>
    </Breadcrumb>
  </Router>
</BrowserRouter>
`;
